src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/RegexNodeConverter.cs

```diff
@@ -32,190 +33,283 @@ public RegexNodeConverter(SymbolicRegexBuilder<BDD> builder, CultureInfo culture
             _captureSparseMapping = captureSparseMapping;
         }
 
-        /// <summary>Converts a <see cref="RegexNode"/> into its corresponding <see cref="SymbolicRegexNode{S}"/>.</summary>
-        /// <param name="node">The node to convert.</param>
-        /// <param name="tryCreateFixedLengthMarker">Whether we should attempt to create a fixed length marker after this node.</param>
-        /// <returns>The generated <see cref="SymbolicRegexNode{S}"/> that corresponds to the supplied <paramref name="node"/>.</returns>
-        public SymbolicRegexNode<BDD> ConvertToSymbolicRegexNode(RegexNode node, bool tryCreateFixedLengthMarker)
+        /// <summary>Converts the root <see cref="RegexNode"/> into its corresponding <see cref="SymbolicRegexNode{S}"/>.</summary>
+        /// <param name="root">The root node to convert.</param>
+        /// <returns>The generated <see cref="SymbolicRegexNode{S}"/> that corresponds to the supplied <paramref name="root"/>.</returns>
+        internal SymbolicRegexNode<BDD> ConvertToSymbolicRegexNode(RegexNode root)
```

Comment:
Talked with Margus and indeed the stack guarding is expensive when hit. We might need to validate places in code that might hit that frequently; probably would be nice to avoid that being hit e.g. on every derivative construction call for even deep patterns to keep the constants down.

Reply:
Certainly hopping to a new thread adds significant expense... but what patterns cause that to be hit?  The theory here was it'd require patterns that essentially don't happen in real-world use.  Is that not the case?

Reply:
The cases in stress tests are clearly possible by a malicious user but hardly useful in any other sense. The new conversion is now guaranteed to be linear in total number of AST nodes in RegexNode and is nonrecursive.